site/_js/components/petitions.js

```diff
@@ -230,14 +232,18 @@ window.components.petitions = function (doc, win) {
         formData.append('member[street_address]', doc.getElementById('form-street_address').value);
       }
 
-      if (doc.getElementById('form-first_name')) {
-        formData.append('member[first_name]', doc.getElementById('form-first_name').value);
-      }
-
       if (doc.getElementById('form-comments')) {
         formData.append('action_comment', doc.getElementById('form-comments').value);
       }
 
+      if (queryString.source) {
+        formData.append('subscription[source]', queryString.source);
+      }
+
+      if (queryString.can_id) {
```

Comment:
Just one more field to be added in mothership—i believe this will accurately track actions from an email link click to a signature. the catch is… idk if the API will actually accept them. (this is the field that's passed on when the page is POSTed from action network to action network)